lib/logger/lib/logger/watcher.ex

```diff
@@ -28,18 +26,24 @@ defmodule Logger.Watcher do
   Removes the given handler.
   """
   def unwatch(mod, handler) do
-    GenEvent.remove_handler(mod, handler, :ok)
+    result = :gen_event.delete_handler(mod, handler, :ok)
+    _ = Supervisor.terminate_child(@name, {__MODULE__, {mod, handler}})
+    result
```

Comment:
We would want to return the result of terminate_child as delete_child could fail if `add_backend` has been called or the Supervisor restarts in between the calls:
```elixir
case Supervisor.terminate_child(@name, child_id) do
  :ok ->
     _ = Supervisor.delete_child(@name, child_id)
    :ok
  {:error, _} = error ->
    error
end
```